coot/moorhen-wrappers.cc

```diff
@@ -50,9 +50,174 @@
 #include "matrix.h"
 
 #include "smilestopdb.h"
-
 using namespace emscripten;
 
+// PRIVATEER FUNCTIONS
+#include <privateer-lib.h>
+#include <privateer-json.h>
+
+struct TorsionEntry {
+  std::string sugar_1;
+  std::string sugar_2;
+  std::string atom_number_1;
+  std::string atom_number_2;
+  float phi;
+  float psi;
+};
+
+struct TableEntry
+{
+  std::string svg;
+  std::string wurcs;
+  std::string chain;
+  std::string glyconnect_id = "NotFound";
+  std::string glytoucan_id = "NotFound";
+  std::string id;
+  int torsion_err = 0;
+  int conformation_err = 0;
+  int anomer_err = 0;
+  int puckering_err = 0;
+  int chirality_err = 0;
+
+  std::vector<TorsionEntry> torsions;
+};
+
+std::vector<TableEntry> validate(const std::string &file, const std::string &name)
```

Comment:
Thanks, the function `privateer_validate` is exactly what I had in mind. But instead of defining `validate` here, can we keep it in `privateer-wrappers.cc` and then import it with `include` to use it here? The `struct` definitions and the bindings at the end of the file is something we may need to keep here though, but that's fine.